kernel/common/float.rb

```diff
@@ -285,4 +285,32 @@ def floor
     return int if self > 0 or self == int
     return int - 1
   end
+
+  def next_float
+    return NAN if self.nan?
+    return -MAX if self == -INFINITY
+    return INFINITY if self == MAX
+    return Math.ldexp 0.5, MIN_EXP - MANT_DIG + 1 if self.zero?
```

Comment:
Explicit parenthesis are preferred here as it would help readability:

    return Math.ldexp(0.5, MIN_EXP - MANT_DIG + 1) if self.zero?